### PR TITLE
fix(node): add nul byte to statfs path on windows

### LINKED ISSUE #22897: `node_unit_tests::_fs_statfs_test` is flaky on windows
https://github.com/denoland/deno/actions/runs/8268046545/job/22619967572?pr=22647#step:39:3153
```
[_fs/_fs_statfs_test 031.70] fs.statfsSync() => ./tests/unit_node/_fs/_fs_statfs_test.ts:53:6
[_fs/_fs_statfs_test 031.70] error: NotFound: The system cannot find the file specified. (os error 2)
[_fs/_fs_statfs_test 031.70]     const statFs = fs.statfsSync(filePath);
[_fs/_fs_statfs_test 031.70]                       ^
[_fs/_fs_statfs_test 031.70]     at Module.statfsSync (ext:deno_node/_fs/_fs_statfs.js:41:18)
[_fs/_fs_statfs_test 031.70]     at fn (file:///D:/a/deno/deno/tests/unit_node/_fs/_fs_statfs_test.ts:56:23)
[_fs/_fs_statfs_test 031.70] 
[_fs/_fs_statfs_test 031.70]  FAILURES 
```

### PR DESCRIPTION
**Pull request by nayeemrmn**
Tentatively closes #22897.

### CODE REVIEW

File: ext/node/ops/fs.rs [hunk 1]
```diff
@@ -185,16 +185,24 @@ where
...
+        GetDiskFreeSpaceW(
+          root.as_ptr(),
+          &mut sectors_per_cluster,
+          &mut bytes_per_sector,
+          &mut available_clusters,
+          &mut total_clusters,
+        )
+      };
+      retries += 1;
+    }
```
Comment (dsherret) [inline ext/node/ops/fs.rs:205]: Libuv seems to not retry on a certain error. Maybe we should do the same? https://github.com/libuv/libuv/blob/fa6745b4f26470dae5ee4fcbb1ee082f780277e0/src/win/fs.c#L2719

File: ext/node/ops/fs.rs [hunk 2]
```diff
@@ -185,16 +185,24 @@ where
...
-        &mut bytes_per_sector,
-        &mut available_clusters,
-        &mut total_clusters,
-      )
-    };
+    let mut code = 0;
+    let mut retries = 0;
+    while code == 0 && retries < 10 {
+      if retries > 0 {
+        std::thread::sleep(std::time::Duration::from_millis(1));
```
Comment (dsherret) [inline ext/node/ops/fs.rs:192]: I believe this will actually sleep for ~15ms on windows. Maybe we should just copy exactly what libuv does?

**Review by dsherret — APPROVED** [review]
LGTM. Thanks!

File: ext/node/ops/fs.rs (hunk 1 above)
Comment (nayeemrmn) [reply ext/node/ops/fs.rs:205]: In our case the canonicalize call above will catch the normal error cases, so imo it's okay if we don't filter the error or have a slightly longer timeout

**Review by dsherret — APPROVED** [review]
LGTM again!

**Pull request merged**